src/main/java/net/dv8tion/jda/api/entities/Message.java

```diff
@@ -483,6 +483,28 @@ default boolean isFromGuild()
      */
     boolean isWebhookMessage();
 
+    /**
+     * If this message is from an application-owned {@link net.dv8tion.jda.api.entities.Webhook Webhook} or
+     * is a response to an {@link net.dv8tion.jda.api.interactions.Interaction Interaction}, this will return
+     * the application's id.
+     * 
+     * @return The application's id or "0" if this message was not sent by an application
+     */
+    @Nonnull
+    default String getApplicationId()
+    {
+        return Long.toUnsignedString(getApplicationIdLong());
+    }
```

Comment:
I feel like our pattern here is normally `id` or `null`

Reply:
This matches the behavior of `MessageChannel#getLatestMessageId`, `MessageReference#getMessageId`, `MessageReference#getChannelId` and `MessageReference#getGuildId`. So I guess those should be changed as well? Maybe in a different PR?

Reply:
> MessageReference#getMessageId

A message will always have an Id. The docs on this method are wrong. It being `"0"` would be a bug.

> MessageReference#getChannelId

A message will always have a channel. The docs on this method are wrong. It being `"0"` would be a bug.

> MessageReference#getGuildId

This should be returning `null` if `getGuildIdLong == 0`, so yeah we need to update it.

Reply:
I'm pretty sure we explicitly decided to use "0" for `getLatestMessageId`.

Reply:
Why?

Reply:
I just looked at the git history:
Originally this method threw an IllegalStateException when you called it if there wasn't a latestMessageId to use. We changed it to instead use `String.valueOf(id)` and later `Long.toUnsignedString(id)`.

I don't see anything in the history that indicates to me that we chose to explicitly do `"0"`

Reply:
I don't remember where or when we talked about it, but I definitely remember discussing it when we removed the exception. And I don't really see a valid reason to change it now. Is there any example of where we *do* return null for missing id?

Reply:
![image](https://user-images.githubusercontent.com/1479909/203121357-5d728ea2-f77c-4bdc-a00f-296e6d285954.png)
These at a minimum

Reply:
> > MessageReference#getMessageId
> 
> A message will always have an Id. The docs on this method are wrong. It being `"0"` would be a bug.
> 
> > MessageReference#getChannelId
> 
> A message will always have a channel. The docs on this method are wrong. It being `"0"` would be a bug.

According to the [docs](https://discord.com/developers/docs/resources/channel#message-reference-object-message-reference-structure) they are optional

Reply:
Those docs are for sending, not receiving.